teos/src/config.rs

```diff
@@ -242,6 +243,60 @@ impl Config {
     pub fn is_default(&self) -> bool {
         self == &Config::default()
     }
+
+    /// checks non-default options using HashMap
+    pub fn non_default_options(&self) -> HashMap<String, (String, String)> {
```

Comment:
nit: I would prepend this function name with `get_`

Reply:
This makes more sense, however, you have not covered all possible options.

Reply:
Everything that is an option can be changed either via cmd or config file, so all options need to be checked.